File: backend/app/schemas/knowledge.py
from datetime import datetime

from pydantic import BaseModel


class KnowledgeDocumentOut(BaseModel):
    id: int
    title: str
    source_type: str
    file_path: str | None
    status: str
    version: int
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class ParsedSectionOut(BaseModel):
    title: str
    content_preview: str
    metadata: dict[str, str]


class ParsedDocumentOut(BaseModel):
    title: str
    file_type: str
    section_count: int
    char_count: int
    sections: list[ParsedSectionOut]


class KnowledgeChunkOut(BaseModel):
    id: int
    document_id: int
    spot_id: int | None
    chunk_index: int
    content_preview: str
    token_count: int

File: frontend/src/types/knowledge.ts
export interface KnowledgeDocument {
  id: number
  title: string
  source_type: string
  file_path: string | null
  status: string
  version: number
  created_at?: string | null
  updated_at?: string | null
}

export interface ParsedSection {
  title: string
  content_preview: string
  metadata: Record<string, string>
}

export interface ParsedDocument {
  title: string
  file_type: string
  section_count: number
  char_count: number
  sections: ParsedSection[]
}

export interface ChunkBuildResponse {
  chunk_count: number
  chunks: Array<{
    id: number
    document_id: number
    spot_id: number | null
    chunk_index: number
    content_preview: string
    token_count: number
  }>
}
